src/gui/shell.cpp

```diff
@@ -93,44 +94,69 @@ QString Shell::fontDesc()
 	return fdesc;
 }
 
-/**
- * Set the GUI font, or display the current font
- */
+/// Set the GUI font, or display the current font
 bool Shell::setGuiFont(const QString& fdesc, bool force)
 {
-	QStringList attrs = fdesc.split(':');
-	if (attrs.size() < 1) {
-		m_nvim->api0()->vim_report_error("Invalid font");
+	// Exit early if the font description does not change, prevent loops.
+	if (fdesc.compare(fontDesc(), Qt::CaseInsensitive) == 0) {
 		return false;
 	}
 
-	qreal pointSize = font().pointSizeF();
-	int weight = -1;
-	bool italic = false;
-	foreach(QString attr, attrs) {
-		if (attr.size() >= 2 && attr[0] == 'h') {
-			bool ok = false;
-			qreal height = attr.mid(1).toFloat(&ok);
-			if (!ok) {
-				m_nvim->api0()->vim_report_error("Invalid font height");
-				return false;
+	bool setShellFontSuccess{ false };
+
+	if (fdesc == "*") {
+		bool fontDialogSuccess = false;
+		QFont font{ QFontDialog::getFont(&fontDialogSuccess, QWidget::font(), this, {},
```

Comment:
This is the modal variant right?

I assume this blocks all event handling (resizes and painting).

Reply:
QWidget::font() could be the current shell font()

Reply:
I think `QWidget::font()` gets the current shell font... Or is there a better way?

Reply:
Interesting, Qt says `QFontDialog` is modal... However the user can still return focus to MainWindow before closing the dialog?

This results in weird behavior, and should be fixed...
![image](https://user-images.githubusercontent.com/11207308/65356373-548a5100-dbc2-11e9-9b7b-9ef68556deaa.png)

Reply:
> I think `QWidget::font()` gets the current shell font... Or is there a better way?

I think my brain is suffering from selective C++ amnesia. I was expecting to just see font() there.